core/storage/user/gae_models.py

```diff
@@ -2400,3 +2403,50 @@ def get_user_id_migration_policy():
         a temporary model, so it doesn't need to be migrated.
         """
         return base_models.USER_ID_MIGRATION_POLICY.NOT_APPLICABLE
+
+
+class PseudonymizedUserModel(base_models.BaseModel):
```

Comment:
Do we have any checks that ensure that pseudonymized user IDs can't be used to "do" things (e.g. write stuff, etc.)? I.e. when handling incoming actions triggered by users, we should presumably verify that the user IDs are regular user IDs and not pseudonymized ones?

Reply:
How do we ensure that it fails, and continues to fail in the future? Is there a backend integration test for that flow?

Reply:
Because there is no `UserSettingsModel` with the ID that starts with `pid_`, there is a prod validation for that and also I have added a domain validation to `UserSettings`.

Reply:
Sorry ... where is this test? I can't find it.